frontend/apps/cloudstorage/webdavapi.lua

```diff
@@ -60,28 +39,30 @@ function WebDavApi:urlEncode(url_data)
     return url_data
 end
 
+-- Append path to address with a slash separator, trimming any unwanted slashes in the process.
+function WebDavApi:getJoinedPath(address, path)
+    local path_encoded = self.urlEncode(path) or ""
+    -- Strip leading & trailing slashes from `path`
+    local sane_path = self.trim_slashes(path_encoded)
+    -- Strip trailing slashes from `address` for now
+    local sane_address = self.rtrim_slashes(address)
+    -- Join our final URL
+    return sane_address .. "/" .. sane_path
+end
+
 function WebDavApi:listFolder(address, user, pass, folder_path, folder_mode)
-    local path = self:urlEncode( folder_path )
+    local path = self.urlEncode(folder_path) or ""
     local webdav_list = {}
     local webdav_file = {}
 
-    local has_trailing_slash = false
-    local has_leading_slash = false
-    if string.sub( address, -1 ) == "/" then has_trailing_slash = true end
-    if path == nil or path == "/" then
-        path = ""
-    elseif string.sub( path, 1, 1 ) == "/" then
-        if has_trailing_slash then
-            -- too many slashes, remove one
-            path = string.sub( path, 2 )
-        end
-        has_leading_slash = true
-    end
-    if not has_trailing_slash and not has_leading_slash then
-        address = address .. "/"
-    end
-    local webdav_url = address .. path
-    if string.sub(webdav_url, -1) ~= "/" then
+    -- Strip leading & trailing slashes from `path`
+    path = self.trim_slashes(path)
+    -- Strip trailing slashes from `address` for now
+    address = self.rtrim_slashes(address)
+    -- Join our final URL, which *must* have a trailing / (it's a URL)
+    -- This is where we deviate from getJoinedPath ;).
+    local webdav_url = address .. "/" .. path
```

Comment:
```suggestion
    local webdav_url = address .. "/" .. self.urlEncode(path)`
```

(So that item_path is *notù URL encoded below)